src/dotnet-roslyn-tools/DartTest/DartTest.cs

```diff
@@ -0,0 +1,246 @@
+            logger.LogError(e, e.Message);
+            return -1;
+        }
+        finally
+        {
+            CleanupDirectory(targetDirectory, logger);
+        }
+
+        return 0;
+    }
+
+    private static async Task<string?> GetLatestShaFromPullRequestAsync(IProduct product, HttpClient gitHubClient, int prNumber, ILogger logger, CancellationToken cancellationToken)
+    {
+        try
+        {
+            var requestUri = $"/repos/dotnet/{product.Name.ToLower()}/pulls/{prNumber}";
+            var response = await gitHubClient.GetAsync(requestUri, cancellationToken).ConfigureAwait(false);
+
+            if (response.IsSuccessStatusCode)
+            {
+                var prData = await response.Content.ReadFromJsonAsync<JsonObject>(cancellationToken: cancellationToken).ConfigureAwait(false);
+                return prData?["head"]?["sha"]?.ToString();
+            }
+            else
+            {
+                logger.LogError($"Failed to retrieve PR data from GitHub. Status code: {response.StatusCode}");
+            }
+        }
+        catch (Exception e)
+        {
+            logger.LogError(e, "Exception while retrieving the latest SHA from GitHub.");
+        }
+
+        return null;
+    }
+
+
+    private static async Task PushPRToInternalAsync(IProduct product, int prNumber, string azureBranchName, ILogger logger, string sha, string targetDirectory, CancellationToken cancellationToken)
+    {
+        try
+        {
+            var initCommand = $"init";
+            await RunGitCommandAsync(initCommand, logger, targetDirectory, cancellationToken).ConfigureAwait(false); ;
+
+            var addGithubRemoteCommand = $"remote add {product.Name.ToLower()} {product.RepoHttpBaseUrl}.git";
+            await RunGitCommandAsync(addGithubRemoteCommand, logger, targetDirectory, cancellationToken).ConfigureAwait(false);
+
+            var addInternalRemoteCommand = $"remote add internal {product.InternalRepoBaseUrl}";
+            await RunGitCommandAsync(addInternalRemoteCommand, logger, targetDirectory, cancellationToken).ConfigureAwait(false);
+
+            var fetchCommand = $"fetch {product.Name.ToLower()} pull/{prNumber}/head";
+            await RunGitCommandAsync(fetchCommand, logger, targetDirectory, cancellationToken).ConfigureAwait(false);
+
+            var checkoutCommand = $"checkout {sha}";
+            await RunGitCommandAsync(checkoutCommand, logger, targetDirectory, cancellationToken).ConfigureAwait(false);
+
+            var checkoutNewBranchCommand = $"checkout -b {azureBranchName}";
+            await RunGitCommandAsync(checkoutNewBranchCommand, logger, targetDirectory, cancellationToken).ConfigureAwait(false);
+
+            var pushCommand = $"push internal {azureBranchName}";
+            await RunGitCommandAsync(pushCommand, logger, targetDirectory, cancellationToken).ConfigureAwait(false);
+        }
+        catch (Exception e)
+        {
+            logger.LogError(e, e.Message);
```

Comment:
Remove this `try catch` as the calling method is already catching exceptions and we should fail if we cannot push the internal branch.